smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolServiceClient.kt

```diff
@@ -38,7 +41,15 @@ open class HttpProtocolServiceClient(
                     }
                     prop.renderInitialization(writer, "config")
                 }
-
+                // Render auth schemes
+                writer.write("var modeledAuthSchemes: [ClientRuntime.AuthScheme] = Array()")
+                if (ServiceIndex(ctx.model).getEffectiveAuthSchemes(ctx.service).contains(SigV4Trait.ID)) {
+                    writer.write("modeledAuthSchemes.append(SigV4AuthScheme())")
+                }
+                if (ServiceIndex(ctx.model).getEffectiveAuthSchemes(ctx.service).contains(SigV4ATrait.ID)) {
+                    writer.write("modeledAuthSchemes.append(SigV4AAuthScheme())")
+                }
```

Comment:
After tinkering with it a bit, I don't think `SwiftIntegration` is the ideal hook to allow auth scheme customization because its main usage rn is to write completely new additional files during codegen (e.g., PaginatorGenerator) or to preprocess models. To allow customers to customize auth schemes used and provide custom auth schemes, I think that would best be handled in followup work to this project which is auth plugin.